### PR TITLE
Split out graphing helper methods

### PR DESCRIPTION
**Pull request by orangejenny**
https://github.com/dimagi/commcare-odk/pull/1277/files/7a5e307a32af5bc318f4b83153c818b03cb84c46#r64393918

@phillipm

### CODE REVIEW

File: app/src/org/commcare/views/EntityDetailView.java [hunk 1]
```diff
@@ -434,4 +369,95 @@ private int getScreenWidth() {
...
     }
+
+    private void addSpinnerToGraph(WebView graphView, ViewGroup graphLayout) {
+        final ProgressBar spinner = new ProgressBar(this.getContext(), null, android.R.attr.progressBarStyleLarge);
+        spinner.setLayoutParams(new FrameLayout.LayoutParams(ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT, Gravity.CENTER));
+        GraphLoader graphLoader = new GraphLoader((Activity) this.getContext(), spinner);
+
+        // Set up interface that JavaScript will call to hide the spinner
+        // once the graph has finished rendering.
+        ((WebView)graphView).addJavascriptInterface(graphLoader, "Android");
```
Comment (phillipm) [inline app/src/org/commcare/views/EntityDetailView.java]: Would be good to get rid of the lint warning by either adding `@SuppressLint("AddJavascriptInterface")` to the method (with a comment as to why it should be safe) or adding code to disable the spinner JS code when running on devices below sdk level 17.

Comment (phillipm) [inline app/src/org/commcare/views/EntityDetailView.java]: `((WebView)graphView)` -> `graphView`

**Comment by phillipm** [comment]
Yay, so much more readable!

File: app/src/org/commcare/views/EntityDetailView.java (hunk 1 above)
Comment (orangejenny) [inline app/src/org/commcare/views/EntityDetailView.java]: > Would be good to get rid of the lint warning by either

Ended up doing both: skipping the method pre-jellybean, and suppressing the warning to make AS happy.

**Pull request merged**